crates/ra_db/src/input.rs

```diff
@@ -122,9 +122,16 @@ pub enum Edition {
     Edition2015,
 }
 
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub struct ExternSourceId(pub u32);
+
 #[derive(Default, Debug, Clone, PartialEq, Eq)]
 pub struct Env {
     entries: FxHashMap<String, String>,
+
+    // Note: Some env variables (e.g. OUT_DIR) are located outside of the
+    // crate. We store a map to allow remap it to ExternSourceId
+    extern_paths: FxHashMap<String, ExternSourceId>,
```

Comment:
Hm, it's ... less then ideal that we need to single out `OUT_DIR`  this way, but I guess that's the only way given current VFS API...

Reply:
Yes, it is the only solution I came up without altering too much code. But after some thoughts, I think I should separate this part to a new struct `ExternPaths` and put it in `CrateGraph` instead.

Reply:
Done!